DetailedSoldierListWOTC/Src/DetailedSoldierListWOTC/Classes/UIPersonnel_SoldierListItemDetailed.uc

```diff
@@ -63,7 +65,18 @@ simulated function string GetPromotionProgress(XComGameState_Unit Unit)
 		return "";
 	}
 
-	promoteProgress = Unit.GetTotalNumKills() $ "/" $ class'X2ExperienceConfig'.static.GetRequiredKills(Unit.GetSoldierRank() + 1);
+	if (SHOW_KILL_XP_NOT_ASSISTS)
+	{
+		promoteProgress = Unit.GetTotalNumKills() $ "/" $ class'X2ExperienceConfig'.static.GetRequiredKills(Unit.GetSoldierRank() + 1);
+	}
+	else
+	{
+		// Deduct the existing assists contribution before multiplying by KillAssistsPerKill.
+		// Ignoring psi credits as they appear not to be used at all.
+		NumKills = Unit.GetTotalNumKills() - Round(Unit.KillAssistsCount) / ClassTemplate.KillAssistsPerKill;
```

Comment:
instead of deducting then adding, you can just multiply both kills and kills required by kill assist per kill then adding assisst modulo kill assist per kill

Reply:
I didn't quite follow what you were saying here, but hope the change I pushed is satisfactory.